model_inference/bird_predictor.py

```diff
@@ -17,32 +15,49 @@
         6: 'Цыпленок'
     }
 
-def model_upload():
-  output = 'bird_recognition.h5'
-  if not os.path.isfile(output):
-    file_id = '1-bWaePLw8owneHJnK2US7jcYnG5gaI5j'
-    url = f'https://drive.google.com/uc?id={file_id}'
-    gdown.download(url, output, quiet=False)
-  model = load_model(output)
-  return model
+def download_file_from_url(url:str, destination_path:str) -> None:
+    response = requests.get(url, stream=True)
+    
+    with open(destination_path, 'wb') as file:
+        for chunk in response.iter_content(chunk_size=128):
+            file.write(chunk)
 
+class Model:
 
-def preprocess_img(image_path):
-  img = image.load_img(image_path, target_size=(224, 224))
-  img_array = image.img_to_array(img)
-  img_array = np.expand_dims(img_array, axis=0)
-  img_array = preprocess_input(img_array)
-  return img_array
+  def __init__(self, weights_path:str) -> None:
+    self.__weights_path = weights_path
+    self.__model = self.__initialize_model()
+  
+  def __download_model_weights(self) -> None:
+    if not os.path.isfile(self.__weights_path):
+      file_url = 'https://drive.google.com/uc?id=1-bWaePLw8owneHJnK2US7jcYnG5gaI5j'
+      download_file_from_url(file_url, self.__weights_path)
 
-def predict(model,img_array):
-  predictions = model.predict(img_array)
-  score = np.max(predictions)
-  kind = np.where(predictions==np.max(predictions))[1]
-  kind=kind[0]
-  return score,kind
+  def __initialize_model(self) -> Model:
+    self.__download_model_weights()
+    return load_model(self.__weights_path)
 
+  def make_predictions(self, img_array:np.ndarray) -> tuple[np.float32, np.int64]:
+    predictions = self.__model.predict(img_array)
+    score = np.max(predictions)
+    kind = np.where(predictions==np.max(predictions))[1]
+    kind=kind[0]
+    return score,kind
```

Comment:
I have to say, I'm a formatting maniac. I have a hard time seeing "=" and "==" without spaces around them, "," without a space after it. I suggest fixing that too, although it's certainly not a bug or anything like that :)

You can spend some more time and try to customize the linking for this project to check codestyle. Suitable plugin: https://pypi.org/project/pylint/

After that, you can try to write a Github Action that will trigger a linting job on every pull-request and prevent the merge pull-request in the master-branch if the linting is unsuccessful.